src/core/ext/transport/chttp2/transport/flow_control.cc

```diff
@@ -296,9 +296,12 @@ FlowControlAction TransportFlowControl::PeriodicUpdate() {
     // target might change based on how much memory pressure we are under
     // TODO(ncteisen): experiment with setting target to be huge under low
     // memory pressure.
-    double target = kSmoothMemoryPressure
-                        ? TargetInitialWindowSizeBasedOnMemoryPressureAndBdp()
-                        : pow(2, SmoothLogBdp(TargetLogBdp()));
+    uint32_t target = static_cast<uint32_t>(RoundUpToPowerOf2(
+        Clamp(kSmoothMemoryPressure
+                  ? TargetInitialWindowSizeBasedOnMemoryPressureAndBdp()
+                  : pow(2, SmoothLogBdp(TargetLogBdp())),
+              0.0, static_cast<double>(kMaxInitialWindowSize))));
```

Comment:
Should Clamp be between 0 and kMaxInitialWindowSize ? Previously it seems to be between kMinInitialWindowSize and kMaxInitialWindowSize. Why is the lower bound 0 now ?

Reply:
`kMinInitialWindowSize` is 0 - which is where it should be. We want resource quota to be able to stop reading from streams, not to stall after some small number of bytes (meaning we need to allocate a little bit of memory for every stream).

Previously we had a bit of a sloppiness factor due to some bugs. I'm going to argue that the fuzzer we have in place is showing good evidence that those bugs no longer remain.